### DISCUSSION TITLE
(Auto-)Switch WiFi mode

### DISCUSSION THREAD

**Discussion by DominicWindisch** [photobooth-app/photobooth-app#177] [Ideas]
I'm currently tinkering with photobooth on a RPI 4, Rpi os bookworm. It's connected to my local wifi.

To "deploy" the physical photobooth I'm planning on using a tablet as user interface connected to the RPI in a direct WiFi connection. I have zero experience with setting that up and I'm very much looking forward to locking myself out of my RPI in the process 😂

Ideally, I'd like to have a button in the UI to switch the RPI between the modes. I.e.:

- button "Switch to Direct WiFi Mode" => start script
- script will switch WiFi mode (somehow) to direct
- button "Switch WiFi to access local network" (possibly even with a way to edit credentials)
- _the important bit_: it will try to connect to the wifi and return to direct mode if it is not successful

This setup would make it rather easy to setup a connection in a headless setup without the danger of locking yourself out. Switching modes may come in handy to provide local access for guests but also have a more convenient interface as admin with internet connection for updates etc. 

Are there already plans on a feature along that lines? Do you have any other feedback on the idea?

[This](https://www.raspberryconnect.com/projects/65-raspberrypi-hotspot-accesspoints/157-raspberry-pi-auto-wifi-hotspot-switch-internet) may be a starting point ([bookworm version](https://www.raspberryconnect.com/projects/65-raspberrypi-hotspot-accesspoints/203-automated-switching-accesspoint-wifi-network)).

**Comment by mgineer85** [comment]
This is a bit out of scope for the app itself and very hard to achieve in a platform agnostic way.
In the past I setup similar like described in your cited website.
At least for bullseye, it was not either/or but both if I remember correct. You could have a direct wifi and if another wifi is in range (your home-wifi) it would connect to that also.

Maybe this would give you a permanent setup with no additional switches needed?

**Reply by DominicWindisch** [reply]
I didn't even think about concurrent connections, didn't think the RPI was capable of that. But I'm totally fine if that's out of scope and add my own solution. :) The linked solutions runs autonomously either way, there's no real need to bloat the app then.
